CRM/Contribute/BAO/Contribution.php

```diff
@@ -176,7 +179,32 @@ public static function add(&$params, $ids = array()) {
     if ($contributionID && $setPrevContribution) {
       $params['prevContribution'] = self::getValues(array('id' => $contributionID), CRM_Core_DAO::$_nullArray, CRM_Core_DAO::$_nullArray);
     }
-
+    // CRM-18573
```

Comment:
I would think this chunk of code is long enough to get it's own function (with a comment block to explain it's purpose)